### ISSUE TITLE
have a clear folders layout

### ISSUE THREAD

**Issue by or-shachar** [bazelbuild/rules_scala#160]
Hi,

As a new comer to this project I failed to find the logic behind the folders structure, which makes it harder to contribute.

If there is a logic behind the current folder structure - let's document it and publish it in README.
Else -  I suggest finding a good standard layout, ie:
```
.
+-- [src]
+-- [test]
|   +-- [unit]
|   +-- [integration]
|   +-- [scripts] (?)
+-- [libs] / [includes]
+-- [examples]
```

And reorder the project accordingly

**Comment by johnynek** [comment]
Yeah, this has somewhat grown organically and gone through a few phases.

The latest thinking, I think, is to use src and test and to use language at the top level `src/java` or `src/scala` and then standard naming of path based on package.

It would be fine to do the moves to standardize that if you are interested.

**Comment by or-shachar** [comment]
I see there is actually a suggested layout in bazel docs:
https://bazel.build/versions/master/docs/skylark/deploying.html#what-a-rule-repository-should-contain

**Comment by johnynek** [comment]
It does not mention where to put non-skylark code that supports the rules,
which at this point is the bulk of our code.
